eth_account/account.py

```diff
@@ -361,50 +360,38 @@ def setKeyBackend(self, backend):
 
     @combomethod
     def sign_message(self, signable_message: SignableMessage, private_key):
-        #TODO document
-        '''
+        r'''
         Sign the provided message.
 
         This API supports any messaging format that will encode to EIP-191 messages.
 
-        If you would like compatibility with
+        If you would like historical compatibility with
         :meth:`w3.eth.sign() <web3.eth.Eth.sign>`
-        you can use :meth:`~eth_account.messages.defunct_hash_message`.
+        you can use :meth:`~eth_account.messages.encode_defunct`.
 
-        Several other message standards are proposed, but none have a clear
-        consensus. You'll need to manually comply with any of those message standards manually.
+        Other options are the "validator", or "structured data" standards. (Both of these
+        are in *DRAFT* status currently, so be aware that the implementation is not
+        guaranteed to be stable).
 
-        :param message_hash: the 32-byte message hash to be signed
-        :type message_hash: hex str, bytes or int
+        :param signable_message: the encoded message for signing
         :param private_key: the key to sign the message with
         :type private_key: hex str, bytes, int or :class:`eth_keys.datatypes.PrivateKey`
-        :returns: Various details about the signature - most
-          importantly the fields: v, r, and s
+        :returns: Various details about the signature - most importantly the fields: v, r, and s
         :rtype: ~eth_account.datastructures.AttributeDict
 
         .. code-block:: python
 
             >>> msg = "I♥SF"
-            >>> from eth_account.messages import defunct_hash_message
-            >>> msghash = defunct_hash_message(text=msg)
-            HexBytes('0x1476abb745d423bf09273f1afd887d951181d25adc66c4834a70491911b7f750')
+            >>> from eth_account.messages import encode_defunct
+            >>> msghash = encode_defunct(text=msg)
+            SignableMessage(version=b'E', header=b'thereum Signed Message:\n6', body=b'I\xe2\x99\xa5SF')
```

Comment:
Should we maybe add a note about where that version "number" comes from?  Seems like it could cause confusion or people might incorrectly think there's a typo in the documentation.